src/services/mongoose/wallet-invoices.ts

```diff
@@ -49,5 +53,26 @@ export const WalletInvoicesRepository = (): IWalletInvoicesRepository => {
     }
   }
 
-  return { persist, findByPaymentHash }
+  async function* findWalletsWithPendingInvoices():
+    | AsyncGenerator<WalletId>
+    | RepositoryError {
+    let pending
+    try {
+      // select distinct user ids from pending invoices
+      pending = InvoiceUser.aggregate([
+        { $match: { paid: false } },
+        { $group: { _id: "$uid" } },
+      ])
+        .cursor({ batchSize: 100 })
+        .exec()
+    } catch (error) {
+      return new RepositoryError(error)
+    }
+
+    for await (const { _id } of pending) {
```

Comment:
Can't getting the next element ever throw?

Reply:
the return pattern only works in in the first part, this is a generator function and return a cursor, if that fails after open the cursor then will throw an exception